packages/service-library/tests/test_logging_utils_filter.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=redefined-outer-name
+
+import logging
+
+import pytest
+from servicelib.logging_utils import GeneralLogFilter
+
+
+@pytest.fixture
+def logger_with_filter():
+    # Set up a logger for testing
+    logger = logging.getLogger("uvicorn.access")
+    logger.setLevel(logging.DEBUG)
+
+    # Create a list to capture log outputs
+    log_capture = []
+
+    # Create a handler that appends log messages to the log_capture list
+    class ListHandler(logging.Handler):
+        def emit(self, record):
+            log_capture.append(self.format(record))
+
+    handler = ListHandler()
+    logger.addHandler(handler)
+
+    # Set up the filter based on the new logic
+    filtered_routes = [
+        '"GET / HTTP/1.1" 200',
+        '"GET /metrics HTTP/1.1" 200',
+    ]
+
+    # Add the GeneralLogFilter to the logger
+    log_filter = GeneralLogFilter(filtered_routes)
+    logger.addFilter(log_filter)
+
+    # Return logger and the log_capture for testing
+    yield logger, log_capture
+
+    # Cleanup: remove handlers and filters after test
+    logger.handlers.clear()
+    logger.filters.clear()
+
+
+def test_log_filtered_out(logger_with_filter):
+    logger, log_capture = logger_with_filter
+
+    # Create a log record that should be filtered out (matches the filter criteria)
+    record = logger.makeRecord(
+        name="uvicorn.access",
+        level=logging.INFO,
+        fn="testfile",
+        lno=10,
+        msg='"GET / HTTP/1.1" 200 OK',
+        args=(),
+        exc_info=None,
+    )
+    logger.handle(record)
+
+    # Assert no log messages were captured (filtered out)
+    assert len(log_capture) == 0
+
```

Comment:
actually you could benchmark here and compare if the logger performance got impeeded.
see [https://github.com/ionelmc/pytest-benchmark](https://github.com/ionelmc/pytest-benchmark) we use it in some places in the repo already.